### ISSUE TITLE
`tye run` attempts to run test projects

### ISSUE THREAD

**Issue by stevejgordon** [dotnet/tye#206]
`tye run` on a solution with an API, web frontend and a test project, attempts to run the test project. Would it be possible to detect test projects and exclude them?

I worked around this by using `tye init` to create the yaml file and then removed the test project.

**Comment by davidfowl** [comment]
I wonder if we can detect a test project...

**Comment by jkotalik** [comment]
Yeah I started brainstorming about this, and I actually got quite nervous about detecting test projects. I think if we want `tye run` to work without a tye.yaml file, we need to figure out a way to exclude test projects, but adding heuristics for checking if a project is a test project (ex folder path includes "test", project name includes "test", etc.), it can lead to false positives. 

TL;DR I think we _need_ to fix this, but not sure how to sniff for test projects, as David suggests.

**Comment by rynowak** [comment]
We can check for the presence of the test SDK, I think. I'm sure there's an MSBuild way to do this, and *I have the power of MSBuild and anime on my side*.

**Comment by rynowak** [comment]
Ok so the right way to detect this is with `IsTestProject` = true. That's set by the test SDK.

The thing we might want to *instead* is look for projects that produce an EXE. Since we're no longer looking for launch settings we will now also trigger on class libraries.

**Comment by davidfowl** [comment]
> Since we're no longer looking for launch settings we will now also trigger on class libraries.

Eeek

**Comment by ArieJones** [comment]
While I see the IsTestProject in the Tye E2ETest project I don't see it in projects that we have recently upgraded to 3.1 ... for example 

```<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>
    <IsPackable>false</IsPackable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="AutoMapper" Version="9.0.0" />
    <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="7.0.0" />
    <PackageReference Include="FakeItEasy" Version="6.0.0-beta.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="3.1.0" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0-preview-20191115-01" />
    <PackageReference Include="Shouldly" Version="4.0.0-beta0002" />
    <PackageReference Include="xunit" Version="2.4.1" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\src\PromptTime.Domain\PromptTime.Domain.csproj" />
    <ProjectReference Include="..\..\src\PromptTime.Messaging\PromptTime.Messaging.csproj" />
  </ItemGroup>

</Project>```

Shouldn't it handle these cases as well? Microsoft.NET.Test.SDK may very well be the fallback check possibly.... 

I do like the "Drats! 'run' failed" message when using an unsupported file type .. like sqlproj :)

**Comment by rynowak** [comment]
@ArieJones - I think we're talking about two different things. The package reference to `Microsoft.NET.Test.Sdk` will set `IsTestProject` to true whether or not you manually set that property in your project.

Due to complicated build-system reasons we have to manually set `IsTestProject` in our project file.

The way Tye interacts with MSBuild is that we load up the object model, run some basic targets and then read the property values. So we get to observe the effect of lots of processing. We don't scrape the XML.

**Comment by ArieJones** [comment]
Ohhhh ok cool. Bouncing back and forth between current upgrade project,  I am looking at trying Tye+Dapr with to see what it looks like vs Tye source.. and playing the "one of these things is not like the other" game. 

Thanks for the clarification!

**Comment by rynowak** [comment]
PR is up

**Comment by rynowak** [comment]
Some other food for thought - this means that things like console projects and winforms projects (if they are in the solution) will be included. We can't really tell the difference between a worker and a regular console app (by design). So I guess the solution for that case is to have a `tye.yaml`. 

This doesn't feel like something that can ever be made perfect, but getting rid of class libraries and unit tests projects - which we know are going to be roommates with webapps seems important.
